hoomd/hpmc/tune/nec_chain_time.py

```diff
@@ -0,0 +1,286 @@
+            target=OnlyTypes(float,
+                             postprocess=self._process_chain_time_target),
+            solver=SolverStep,
+            max_chain_time=OnlyTypes(
+                float,
+                allow_none=True,
+                postprocess=self._process_chain_time_range),
+            min_chain_time=OnlyTypes(
+                float, postprocess=self._process_chain_time_range))
+
+        self._param_dict.update(param_dict)
+        self.target = target
+        self.solver = solver
+        self.max_chain_time = max_chain_time
+        self.min_chain_time = self._min_chain_time
+
+    def _process_chain_time_range(self, target):
+        # check range
+        if 0 <= target:
+            pass
+        else:
+            raise ValueError(
+                "Chain time {} should be between 0 and 1000.".format(target))
+        self._tuned = 0
+        return target
+
+    def _process_chain_time_target(self, target):
+        # check range
+        if 0 <= target <= 1000:
+            pass
+        else:
+            raise ValueError(
+                "Value {} should be between 0 and 1000.".format(target))
+
+        self._chain_time_def.target = target
+        self._tuned = 0
+        return target
+
+    def attach(self, simulation):
+        if not isinstance(simulation.operations.integrator, HPMCNECIntegrator):
+            raise RuntimeError(
+                "ChainTimeTuner can only be used in HPMC-NEC simulations.")
+        self._chain_time_def.integrator = simulation.operations.integrator
+        self._is_attached = True
+
+    @property
+    def _attached(self):
+        """bool: Whether or not the tuner is attached to a simulation."""
+        return self._is_attached
+
+    @property
+    def tuned(self):
+        """bool: Whether or not the move sizes are considered tuned.
+
+        A `ChainTime` object is considered tuned if it the solver tolerance has
+        been met by all tunables for 2 iterations.
+        """
+        return self._tuned >= 2
+
+    def detach(self):
+        self._chain_time_def.integrator = None
+        self._is_attached = False
+
+    def act(self, timestep=None):
+        """Tune chain time.
+
+        Args:
+            timestep (`int`, optional): Current simulation timestep. Is
+                currently ignored.
+        """
+        if self._is_attached:
+            # update maximum chain time
+            if self._update_chain_time:
+                self._chain_time_def.domain = (self.min_chain_time,
+                                               self.max_chain_time)
+
+            tuned = self.solver.solve([self._chain_time_def])
+            self._tuned = self._tuned + 1 if tuned else 0
+
+
+class ChainTime(_InternalCustomTuner):
+    """Tunes HPMCNECIntegrator chain time to targeted mean particles per chain.
+
+    For most common creation of a `ChainTime` tuner see
+    `ChainTime.secant_solver` and `ChainTime.scale_solver` respectively.
+
```

Comment:
I would also provide a recommended `target` value in the documentation.

Reply:
Good suggestion. This method isn't widely used yet so there may be no good value to suggest. Users will need to refer to the paper to gain any insight in how to use the method effectively.